### ISSUE TITLE
Hardcoded Mask

### ISSUE THREAD

**Issue by majurgens** [nichtich/File-Hotfolder#9]
Don't these lines (113-115):
   unless ( $self->inotify->watch( 
        $path, 
        IN_CREATE | IN_CLOSE_WRITE | IN_MOVE | IN_DELETE | IN_DELETE_SELF | IN_MOVE_SELF, 

mean that the mask is hardcoded?

I don't quite understand why the module seems to filter them post-notifications.
lines 131-132:
         } elsif ( $e->mask & $self->{event_mask} ) {
                $self->_callback($path);

Wouldn't it be better to just pass the configured mask to inotify in the first place?

This would have the additional benefit of being able to specify other inotify settings such as IN_DONT_FOLLOW  etc

**Comment by nichtich** [comment]
Yes, the mask is hard-coded internally to catch all events. It's easier to ignore afterwards. If you miss a feature such as IN_DONT_FOLLOW, please specify what's missing but please don't try to improve the code unless there is an actual bug.
